components/iotserver-api/org.wso2.carbon.device.mgt.iot.common.api/src/main/java/org/wso2/carbon/device/mgt/iot/common/api/PolicyManagementService.java

```diff
@@ -0,0 +1,508 @@
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@PUT
+	@Produces("application/json")
+	@Path("/inactivate/{id}")
+	public boolean inactivatePolicy(@PathParam("id") int policyId) {
+
+		try {
+			PolicyManagerService policyManagerService = getPolicyServiceProvider();
+			PolicyAdministratorPoint pap = policyManagerService.getPAP();
+			pap.inactivatePolicy(policyId);
+			if (log.isDebugEnabled()) {
+				log.debug("Policy by id:" + policyId + " has been successfully inactivated.");
+			}
+			return true;
+		} catch (PolicyManagementException e) {
+			String error = "Exception in inactivating policy by id:" + policyId;
+			log.error(error, e);
+			return false;
+		} catch (DeviceManagementException e) {
+			String error = "Error occurred while invoking Policy Management Service.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@PUT
+	@Produces("application/json")
+	@Path("/apply-changes")
+	public boolean applyChanges() {
+
+		try {
+			PolicyManagerService policyManagerService = getPolicyServiceProvider();
+			PolicyAdministratorPoint pap = policyManagerService.getPAP();
+			pap.publishChanges();
+			if (log.isDebugEnabled()) {
+				log.debug("Changes have been successfully updated.");
+			}
+			return true;
+		} catch (PolicyManagementException e) {
+			String error = "Exception in applying changes.";
+			log.error(error, e);
+			return false;
+		} catch (DeviceManagementException e) {
+			String error = "Error occurred while invoking Policy Management Service.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@GET
+	@Path("/start-task/{milliseconds}")
+	public boolean startTaskService(@PathParam("milliseconds") int monitoringFrequency) {
+
+		try {
+			PolicyManagerService policyManagerService = getPolicyServiceProvider();
+			TaskScheduleService taskScheduleService = policyManagerService.getTaskScheduleService();
+			taskScheduleService.startTask(monitoringFrequency);
+			if (log.isDebugEnabled()) {
+				log.debug("Policy monitoring service started successfully.");
+			}
+			return true;
+		} catch (PolicyMonitoringTaskException e) {
+			String error = "Policy Management related exception.";
+			log.error(error, e);
+			return false;
+		} catch (DeviceManagementException e) {
+			String error = "Error occurred while invoking Policy Management Service.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@GET
+	@Path("/update-task/{milliseconds}")
+	public boolean updateTaskService(@PathParam("milliseconds") int monitoringFrequency) {
```

Comment:
why do not we simply return the javax.ws.rs.core.Response object here as well and make the return type consistent across all the APIs?